moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/parameterization/joint_space/constrained_planning_state_space_factory.h

```diff
@@ -58,7 +58,7 @@ class ConstrainedPlanningStateSpaceFactory : public ModelBasedStateSpaceFactory
    * For more details on this state space selection process, see:
    * https://github.com/JeroenDM/moveit/pull/2
    * **/
-  int canRepresentProblem(const std::string& group, const moveit_msgs::msg::MotionPlanRequest& req,
+  int canRepresentProblem(const std::string& group, const planning_interface::MotionPlanRequest& req,
```

Comment:
I like the idea of all these functions now taking a MoveIt! struct, as opposed to a ROS message. Even though the struct is still inheriting from the message, this can be a step towards additional ROS independency.
Not to do now, but what would you think of a `planning_interface::MotionPlanRequest` struct totally independent from the ROS message, with separate to/from functions to convert between one or the other?